convert2rhel/logger.py

```diff
@@ -299,33 +268,63 @@ def disable_colors(self, value):
         self.color_disabled = value
 
     def format(self, record):
-        if record.levelno == LogLevelTask.level:
-            temp = "*" * (90 - len(record.msg) - 25)
-            fmt_orig = "\n[%(asctime)s] %(levelname)s - [%(message)s] " + temp
-            new_fmt = fmt_orig if self.color_disabled else colorize(fmt_orig, "OKGREEN")
-            self._fmt = new_fmt
+        """Format tasks, etc
+
+        :param logging.LogRecord record: Logger-provided LogRecord that is
+        provided when we use logging.warning() etc.
+        :return str: The formatted log message
+        """
+        fmt_orig = "[%(asctime)s] %(levelname)s - %(message)s"  # DEBUG default
+        self.datefmt = "%Y-%m-%dT%H:%M:%S%z"  # DEBUG default
+
+        # Used when providing isTask in the extra field
+        # e.g. logging.warning("Testing", extra={"isTask": True})
+        isTask = getattr(record, "isTask", False)
+
+        color = self._getLogLevelColor(record, isTask)
+        if isTask:
+            asterisks = "*" * (90 - len(record.msg) - 25)
+            fmt_orig = "\n[%(asctime)s] TASK - [%(message)s] " + asterisks
             self.datefmt = "%Y-%m-%dT%H:%M:%S%z"
-        elif record.levelno in [logging.INFO]:
-            self._fmt = "%(message)s"
-            self.datefmt = ""
-        elif record.levelno in [logging.WARNING]:
-            fmt_orig = "%(levelname)s - %(message)s"
-            new_fmt = fmt_orig if self.color_disabled else colorize(fmt_orig, "WARNING")
-            self._fmt = new_fmt
-            self.datefmt = ""
-        elif record.levelno >= logging.ERROR:
-            # Error, Critical, Critical_no_exit
+        elif record.levelno >= logging.WARNING:
             fmt_orig = "%(levelname)s - %(message)s"
-            new_fmt = fmt_orig if self.color_disabled else colorize(fmt_orig, "FAIL")
-            self._fmt = new_fmt
+        elif record.levelno >= logging.INFO:
+            fmt_orig = "%(message)s"
             self.datefmt = ""
-        else:
-            self._fmt = "[%(asctime)s] %(levelname)s - %(message)s"
-            self.datefmt = "%Y-%m-%dT%H:%M:%S%z"
 
+        new_fmt = fmt_orig if not color or self.color_disabled else colorize(fmt_orig, color)
+        self._fmt = new_fmt
```

Comment:
What do you think unpacking the if-else one-liner to this? 

In my opinion, it makes easier to read when we have a if with more than one statement like the current one. 

```suggestion
        new_fmt = fmt_orig
        if color or not self.color_disaled:
            new_fmt = colorize(fmt_orig, color)

        self._fmt = new_fmt
```

Not a required change, to be honest, just posting in case if you agree to apply.